File: Check-If-String-Is-a-Prefix-of-Array.py
class Solution(object):
    def isPrefixString(self, s, words):
        """
        :type s: str
        :type words: List[str]
        :rtype: bool
        """
        t = ""
        for word in words:
            t += word
            if t == s:
                return True
            if len(t) > len(s):
                return False
        return False

File: Check-If-String-Is-a-Prefix-of-Array.js
/**
 * @param {string} s
 * @param {string[]} words
 * @return {boolean}
 */
var isPrefixString = function(s, words) 
{
    let temp = "";
    for (let i = 0; i < words.length; i++) 
    {
        temp += words[i];
        if (temp === s) 
        {
            return true;
        }
        if (temp.length > s.length)
        {
            return false;
        } 
    }
    return false;
};
